reserved_keyword_checker/reserved_keyword_checker.py

```diff
@@ -0,0 +1,292 @@
+        if override_file:
+            self.overrides = self.read_config_file(override_file)
+        else:
+            self.overrides = {}
+        self.validate_override_config()
+        self.report_path = report_path
+        self.report_file = os.path.join(report_path, "reserved_keyword_report.csv")
+
+    @staticmethod
+    def read_config_file(config_file_path):
+        click.echo("Loading config file: {}".format(config_file_path))
+        try:
+            with io.open(config_file_path, 'r') as config_file:
+                config_dict = yaml.safe_load(config_file)
+        except:
+            raise ConfigurationException("Unable to load config file: {}".format(config_file_path))
+        if not config_dict:
+            raise ConfigurationException("Config file is empty: {}".format(config_file_path))
+        return config_dict
+
+    def validate_override_config(self):
+        field_path_regex = re.compile(r'([A-Z]\w*\.)(\w+)')
+        for system, override_list in self.overrides.items():
+            for pattern in override_list:
+                if not re.search(field_path_regex, pattern):
+                    raise ConfigurationException("Invalid value in override file: {}".format(pattern))
+
+
+def collect_concrete_models():
+    """
+    Walk through all of the INSTALLED_APPS in a Django project, gathering all
+    of the 'concrete' models. In this case, 'concrete' refers to the fact
+    that a model has a corresponding tables within a database (as opposed to
+    an abstract model, which does not). For more information, see:
+    https://openedx.atlassian.net/wiki/spaces/PLAT/pages/895287378/OEP-30+Implementation
+    """
+    def is_concrete(model):
+        return (
+            issubclass(model, models.Model) and
+            not model is models.Model and
+            not model._meta.abstract and
+            not model._meta.proxy
+        )
+
+    concrete_models = set()
+
+    click.echo("Collecting all concrete models in installed apps")
+    for app in django.apps.apps.get_app_configs():
+        click.echo("Inspecting app: {}".format(app))
+        app_models = []
+        for root_model in app.get_models():
+
+            model_hierarchy = inspect.getmro(root_model)
+            for model in model_hierarchy:
+                if is_concrete(model):
+
+                    model_name = model._meta.object_name
+                    concrete_models.add(model)
+                    app_models.append(model_name)
+        if app_models:
+            click.echo("Found models: {}".format(','.join(app_models)))
+
+    click.echo("Collected {} concrete models".format(len(concrete_models)))
+    return list(concrete_models)
+
+
+def get_fields_per_model(model):
+    """
+    Given a model, return a list of all of the field names on the model,
+    regardless of whether they are explicitly present or present through
+    inheritance. Do not include hidden fields, as these are not created
+    in app code.
+    """
+    return [
```

Comment:
Since what we care about is the database table name and those can be overridden you might want to return the Meta `db_table` here. I'm not sure there is a `db_table` in the case of non-override, we'd have to check.

Reply:
Odd. I thought this would have been marked as `outdated` as modified the code around this comment. Either way, I think I have addressed this.

Reply:
agreed. resolving comment.